jerry-ext/context/context.c

```diff
@@ -0,0 +1,96 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+#include "jerryscript-ext/context.h"
+#include "jmem.h"
+#include "section-utils.h"
+
+static int jerryx_context_slot_count = -1;
+
+/* The section name must not exceed 16 characters on __MACH__, hence the abbreviation "ctx" */
+DECLARE_SECTION (jerryx_ctx_slots, jerryx_context_slot_t)
+
+/**
+ * Create a new context. This function can be passed to jerry_init_with_user_context () as its second parameter.
+ *
+ * @return the pointer to store as the user context.
+ */
+void *
+jerryx_context_init (void)
+{
+  void **ret_pp;
+  int index;
+  size_t context_size;
+  jerryx_context_slot_t *slot_p = NULL;
+
+  if (unlikely (jerryx_context_slot_count < 0))
```

Comment:
is this `unlikely` needed here, is this code performance critical? note that this is a jerry-core internal macro. should we ever be able to get rid of deep-linking to jmem.h, we will lose this, too